### ISSUE TITLE
images didn't cached

### ISSUE THREAD

**Issue by Marwan88** [rs/SDWebImage#2471]
Dears,

I have a problem when cache image in table view or collection view 
the problem is when load first cells the images has been loaded then move scroll down then back to up the images already loaded it's clear and download again, 

there issue i'm not sure if related a problem, when move to down some images return error from cache "canceled" 

to more details please see the [video](https://drive.google.com/open?id=1N4HHyGJRGOWUc2Bc5Cy97ZbzVvfjkA7l) 

Many Thanks
Regards

**Comment by dreampiggy** [comment]
Please fill the issue template instead of removing it...We need to know the basic information to identify the issue. Including SDWebImage version, platform OS version.

And, did you set any configuration in `SDImageCacheConfig` like `maxCacheAge` ? Did you load the image with `FLAnimatedImageView` in 4.x ? It's also recommend to provide a code snippet of your usage or a demo project to reproduce this issue.

**Comment by Marwan88** [comment]
### New Issue Checklist

* [ Yes] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ Yes] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [Yes ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           |  ios
 Platform Version        |  11.4
 SDWebImage Version      | 4.2.2 
 Integration Method      | cocoapods 
 Xcode Version           |  Xcode 9
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.
